enzyme/benchmarks/ReverseMode/fft/src/safe.rs

```diff
@@ -27,37 +31,35 @@ fn bitreversal_perm<T>(data: &mut [T]) {
     }
 }
 
-fn radix2(data: &mut [f64], i_sign: f64, n: usize) {
+fn radix2(data: &mut [f64], i_sign: f64) {
+    let n = data.len() / 2;
     if n == 1 {
         return;
     }
 
-    let (a,b) = data.split_at_mut(n);
-    radix2(a, i_sign, n/2);
-    radix2(b, i_sign, n/2);
+    let (a, b) = data.split_at_mut(n);
+    radix2(a, i_sign);
+    radix2(b, i_sign);
 
     let wtemp = i_sign * (PI / n as f64).sin();
     let wpi = -i_sign * (2.0 * PI / n as f64).sin();
     let wpr = -2.0 * wtemp * wtemp;
     let mut wr = 1.0;
     let mut wi = 0.0;
 
-    assert_eq!(data.len(), 2*n);
-    for i in (0..n).step_by(2)  {
-        let in_n = i + n;
-        let tempr = data[in_n] * wr - data[in_n + 1] * wi;
-        let tempi = data[in_n] * wi + data[in_n + 1] * wr;
+    for i in (0..n).step_by(2) {
+        let tempr = b[i] * wr - b[i + 1] * wi;
+        let tempi = b[i] * wi + b[i + 1] * wr;
 
-        data[in_n] = data[i] - tempr;
-        data[in_n + 1] = data[i + 1] - tempi;
-        data[i] += tempr;
-        data[i + 1] += tempi;
+        b[i] = a[i] - tempr;
+        b[i + 1] = a[i + 1] - tempi;
+        a[i] += tempr;
+        a[i + 1] += tempi;
 
         let wtemp_new = wr;
         wr += wr * wpr - wi * wpi;
         wi += wi * wpr + wtemp_new * wpi;
     }
-
 }
```

Comment:
@ZuseZ4 Let me know if you don't like this. (Other changes in this commit are just rustfmt/whitespace.)